src/ai/backend/common/types.py

```diff
@@ -1474,3 +1474,22 @@ class AutoScalingMetricComparator(CIUpperStrEnum):
     LESS_THAN_OR_EQUAL = enum.auto()
     GREATER_THAN = enum.auto()
     GREATER_THAN_OR_EQUAL = enum.auto()
+
+
+T = TypeVar("T")
+
+
+@dataclass
+class Result(Generic[T]):
+    result: Optional[T] = None
+    error: Optional[str] = None
+    issues: list[str] = field(default_factory=list)
+
+    def is_success(self) -> bool:
+        return self.error is None
+
+    def is_failure(self) -> bool:
+        return self.error is not None
+
+    def has_issues(self) -> bool:
+        return bool(self.issues)
```

Comment:
Result is too general name. 
And wouldn't it be better not to define `issues` if we define `errors`?

Reply:
For now, we have changed this to `MultipleResult`.